services/frontend/src/components/WeatherWidgetContainer.js

```diff
@@ -0,0 +1,42 @@
+import PropTypes from "prop-types";
+import WeatherWidget from "./WeatherWidget";
+
+const WeatherWidgetContainer = ({ weatherObjArray, widgetCount }) => {
+  const createWeatherWidgets = (objArray) => {
+    const widgetArray = [];
+    for (let index = 0; index < widgetCount; index++) {
```

Comment:
lol, this is pretty poorly written and I should add comments.

Basically I'm just validating that there are at least N elements in the weatherObjArray and does validation on each object.

Reply:
Like state, props must not be edited directly inside the component.

Reply:
Shouldn't the count (`widgetCount`) be determined from the `weatherObjArray`?

Reply:
Yeah looking back this isn't very good design. I think I was trying to decouple the array length from the container length (i.e. regardless of the array you passed in the container would either be padded or truncated to the length you specified). But on further consideration I don't see why that would be useful.

We can rewrite this as we nail down the backend.